prov/efa/src/efa_domain.c

```diff
@@ -253,13 +254,11 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 		domain->type = EFA_DOMAIN_RDM;
 		rxr_domain = container_of(domain_fid, struct rxr_domain,
 					  rdm_domain);
-		/*
-		 * If FI_MR_LOCAL is set, we do not want to use the
-		 * MR cache
-		 */
-		enable_cache = efa_mr_cache_enable && !rxr_domain->rxr_mr_local;
+		app_mr_local = rxr_domain->rxr_mr_local;
 	} else {
 		domain->type = EFA_DOMAIN_DGRAM;
+		/* DGRAM always requires FI_MR_LOCAL */
```

Comment:
I am not sure I understand this comment. If app does not specify FI_MR_LOCAL in MR mode, efa dgram will not be picked?

Reply:
Yes, that's correct. The DGRAM endpoint can only support applications that handle their own registrations.

Reply:
I see, thanks!